### PR TITLE
[Bug][flink-connector-elasticsearch7]sink failure caused by deprecated TransportClient

### LINKED ISSUE #2539: [Bug] [flink-connector-elasticsearch7] sink failure caused by unsupported deprecated TransportClient on 7.x
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

flink-connector-elasticsearch7 work fails

### SeaTunnel Version

dev, 2.1.2, 2.1.3

### SeaTunnel Config

```conf
env {}

source {
    JdbcSource {
      driver = com.mysql.cj.jdbc.Driver
      url = "jdbc:mysql://localhost:3306/data_service"
      username = root
      password = 123456
      query = "select * from sample_data_e_commerce"
      result_table_name = sample_data_e_commerce
    }
}

transform {}

sink {
  Elasticsearch {
    source_table_name = sample_data_e_commerce
    hosts = ["localhost:9200"]
  }
}
```

### Running Command

```shell
unneeded
```

### Error Exception

```log
22/08/26 20:39:54 ERROR ElasticsearchOutputFormat: None of the configured nodes are available: [{#transport#-1}{hVAKLFNeSJm2OOzgoLL92Q}{localhost}{127.0.0.1:9200}]
NoNodeAvailableException[None of the configured nodes are available: [{#transport#-1}{hVAKLFNeSJm2OOzgoLL92Q}{localhost}{127.0.0.1:9200}]]
	at org.elasticsearch.client.transport.TransportClientNodesService.ensureNodesAreAvailable(TransportClientNodesService.java:352)
	at org.elasticsearch.client.transport.TransportClientNodesService.execute(TransportClientNodesService.java:248)
	at org.elasticsearch.client.transport.TransportProxyClient.execute(TransportProxyClient.java:57)
	at org.elasticsearch.client.transport.TransportClient.doExecute(TransportClient.java:394)
	at org.elasticsearch.client.support.AbstractClient.execute(AbstractClient.java:396)
	at org.elasticsearch.client.support.AbstractClient.bulk(AbstractClient.java:474)
	at org.elasticsearch.action.bulk.Retry$RetryHandler.execute(Retry.java:205)
	at org.elasticsearch.action.bulk.Retry.withBackoff(Retry.java:59)
	at org.elasticsearch.action.bulk.BulkRequestHandler.execute(BulkRequestHandler.java:62)
	at org.elasticsearch.action.bulk.BulkProcessor.execute(BulkProcessor.java:455)
	at org.elasticsearch.action.bulk.BulkProcessor.internalAdd(BulkProcessor.java:389)
	at org.elasticsearch.action.bulk.BulkProcessor.add(BulkProcessor.java:361)
	at org.elasticsearch.action.bulk.BulkProcessor.add(BulkProcessor.java:347)
	at org.apache.seatunnel.flink.elasticsearch.sink.ElasticsearchOutputFormat$2.add(ElasticsearchOutputFormat.java:116)
	at org.apache.seatunnel.flink.elasticsearch.sink.Elasticsearch.lambda$outputBatch$e5a96c4d$1(Elasticsearch.java:128)
	at org.apache.seatunnel.flink.elasticsearch.sink.ElasticsearchOutputFormat.writeRecord(ElasticsearchOutputFormat.java:134)
	at org.apache.flink.runtime.operators.DataSinkTask.invoke(DataSinkTask.java:240)
	at org.apache.flink.runtime.taskmanager.Task.doRun(Task.java:779)
	at org.apache.flink.runtime.taskmanager.Task.run(Task.java:566)
	at java.lang.Thread.run(Thread.java:748)
```

### Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by kalencaya**
## Purpose of this pull request
fix #2539 

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-elasticsearch7/src/main/java/org/apache/seatunnel/flink/elasticsearch/sink/ElasticsearchOutputFormat.java [hunk 1]
```diff
@@ -64,27 +62,14 @@ public ElasticsearchOutputFormat(Config userConfig, ElasticsearchSinkFunction<T>
     @Override
     public void configure(Configuration configuration) {
         List<String> hosts = config.getStringList(HOSTS);
-        Settings.Builder settings = Settings.builder();
-
-        config.entrySet().forEach(entry -> {
```
Comment (Hisoka-X) [inline seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-elasticsearch7/src/main/java/org/apache/seatunnel/flink/elasticsearch/sink/ElasticsearchOutputFormat.java:69]: Seem like you remove custom setting for elasticsearch, please add it back.

Comment (kalencaya) [reply seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-elasticsearch7/src/main/java/org/apache/seatunnel/flink/elasticsearch/sink/ElasticsearchOutputFormat.java:69]: Elasticsearch will not support settings on rest client, neither 7.x RestHighLevelClient or 8.x ElasticsearchClient, for what all them based on RestClient doesn't support it.

**Comment by Hisoka-X** [comment]
Can you add connector e2e-testcase and update doc

reference：https://github.com/apache/incubator-seatunnel/pull/2499

**Comment by EricJoy2048** [comment]
Since this pr has not been updated for too long, I will close it. If necessary, please reopen it.

**Pull request closed without merging**